### DISCUSSION TITLE
Cannot use RELAY COMPONENT to switch the periscope view.

### DISCUSSION THREAD

**Discussion by hxfmwj** [FakeFishGames/Barotrauma#14069] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Cannot use RELAY COMPONENT to switch the periscope view.
![MO3YYJU$6R$3QCLE%$2_JH2](https://github.com/FakeFishGames/Barotrauma/assets/166732182/98dde685-5b81-40af-a028-a40c25b93cee)

### Reproduction steps

Cannot use RELAY COMPONENT to switch the periscope view.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.0.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by hxfmwj** [comment]
![FVL)@WLPQ D{GX~KCK`Z0HP](https://github.com/FakeFishGames/Barotrauma/assets/166732182/824c4d2a-65ff-4bf0-8cad-02a9bd07b021)
It's not a MOD problem.

**Comment by jcx1991911** [comment]
Well, I just happened to be recording this lol
It seems appeared in version 1.5.3.0, switch with WiFi component still work, maybe it has something to do with the new distance_out pin of periscope
(probably this thing disrupts the logic of the periscope view)

https://github.com/FakeFishGames/Barotrauma/assets/116519004/bf956d97-d8ec-4192-9e42-d5c025ca76ec

**Comment by Regalis11** [comment]
Thank you for the report! [Fixed in our private repo now](https://github.com/FakeFishGames/Barotrauma/issues/14082), the fix should be coming in the next unstable build.

> Fixed toggling between turrets no longer working due to the changes in https://github.com/FakeFishGames/Barotrauma-development/commit/80db478d7e825a9353872e40a51c5d6973c791e4
https://github.com/FakeFishGames/Barotrauma-development/commit/80db478d7e825a9353872e40a51c5d6973c791e4 (new "distance_out" connection). Sending the "distance_out" signal after "position_out" in `GetFocusTarget` meant that the focus target was now determined by "distance_out".
